backstage/templates/backend-deployment.yaml

```diff
@@ -15,6 +15,9 @@ spec:
     metadata:
       annotations:
         ad.datadoghq.com/backstage.logs: '[{"source":"backstage","service":"backend"}]'
+        {{- if .Values.externalSecrets.enabled }}
+        secret.reloader.stakater.com/reload: {{ include "backstage.externalSecretsList" . }}
```

Comment:
This is pretty neat. 
Thinking out loud, and I suppose related to my question below, this reload will/can happen in response to a user entering data through the UI? If that's the case I'm just thinking what UX impact there is restarting the backend. Will any requests fail as we restart the backend (we only run a single node to accommodate techdocs IIRC, though I'm not sure if that actually matters here)? Also, how does the user know when the secret is available? Basically, if they add a secret and immediately try and use it things might not work though I expect that window is kind small (a couple of minutes).
I suppose ideally the frontend could know that the backend is being restarted or that the secret is/isn't yet available to the backend and make this clear to the user.
Again it's not related to this PR as such but more to the corresponding backstage plugin and definitely something we can kick down the line.

Reply:
Very good point. I changed the message here: https://github.com/RoadieHQ/backstage-plugin-secrets-settings/pull/2 (I am happy to change the message or do something else entirely to make sure this is clear to the user)

I will run some tests to see what the effects are on practice. I believe it can take 1-2 mins or even less:
- the external-secrets operator checks for changes every 10 seconds
- the reloader acts almost immediately 
- the backend takes a few seconds to start up
- until the new backend is up and running the old one is still serving requests

Reply:
Sounds good.
I had a thought the other day that - down the line - we could check if a secret is updated. We _could_ check that for a given secret name that the value in the backend process.env matches the value in ssm. If not then we know it's stale. (And we could render an updating icon like AWS does?) 
Anyway, I'm not sure if that or any variation is acceptable from a security standpoint.
Not something we need to implement now but might be pragmatic later.

Reply:
Great idea, this can save a lot of troubleshooting time for us and customers. I added a card here https://app.clubhouse.io/larder/story/860